README.md

```diff
@@ -1,13 +1,104 @@
-erl_optics
-=====
+# erl_optics
 
-An OTP library
+An OTP application for [optics](https://github.com/RAttab/optics).
 
-### Requirements
+## Requirements
+-----
 
 * Erlang OTP >= 21
+* [optics](https://github.com/RAttab/optics) v1.1.9
 
-Build
+## Building
 -----
-
     $ rebar3 compile
+
+## Usage
+-----
+
+For now, the only working mode is Carbon.
+
+need env :
+```
+  {erl_optics, [
+    {hostname, "<address of carbon host>"},
+    {port, <port number>},
+    {mode, carbon},
+    {interval, <polling interval time (ms)>}
+    ]}
+```
+
+To start the application:
+```
+  erl_optics_app:start(),
+```
+
+To start the optics polling:
+```
+  erl_optics:start_optics(<<"<namespace>">> [, <list of lenses>]),
+```
+The namespace will be prepending the name of every metric, for
+example: `"optics_namespace.my_counter"`
+
+The only lenses that require initial declaration are histo and quantile. If
+they are unused, they can be omitted, otherwise, they must be declared like so:
+
+```
+  Lenses = [erl_optics_lens:histo(<<"histo_name">>, [10, 20, 30, 40]),
+            erl_optics_lens:quantile(<<"quantile_name">>, 0.50, 10.0, 0.5)],
+  erl_optics:start_optics(<<"optics_namespace">>, Lenses).
+```
+
+## Lenses
+
+### counter
+A simple counter, polling will give the total increment value. Increment
+defaults to 1.
+```
+  erl_optics:counter_inc(<<"<counter_name>">> [, <increment>]),
+```
+
+### gauge
+Will keep the last value given to it.
+```
+  erl_optics:gauge_set(<<"<gauge_name>">>, <value>),
+```
+
+### dist
+Gives an estimation of the distribution of the values given to it. It provides
+metrics for 50th, 90th and 99th percentiles, the total number of values fed to
+it and the maximum value.
+```
+  erl_optics:dist_record(<<"<dist_name>">>, <value>)
+```
+Dist can also be used to record timings with the two variants
+`dist_record_timing_now` (s) and `dist_record_timing_now_us` (ms) that expect
+erlang timestamps as values and will record the difference between
+the moment it is called and the timestamp receives:
+```
+  Time = os:timestamp(),
+  % some things happen here
+  erl_optics:dist_record_timing_now_us(<<"dist_name">>, Time),
+```
+
+### histo
+Histo is a collection of buckets who are incremented every time a value
+within their boundaries is received. It is necessary to declare these
+lenses when starting optics:
+```
+  erl_optics_lens:histo(<<"<histo_name>">>, <buckets_list>)
```

Comment:
there are technically limitations around the bucket list and how to provide them. That being said, I wouldn't spend too much time on this lens as I'm not sure if it's useful at all.